packages/context/src/provider.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BoundValue} from './binding';
+
+/**
+ * class definition for provider meta definition
+ */
+export class ProviderMeta {
```

Comment:
I think it's a bit over-engineering here. The only critical contract for a `ValueProvider` is that it has `value()` method. It's not the value provider's responsibility to provide a key. A component can return the following object:
```js
{
  key1: ProviderClass1,
  key2: ProviderClass2
}
```